cloudinit/config/cc_netplan_nm_patch.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2024, Raspberry Pi Ltd.
```

Comment:
If I remember correctly, the problem was if you edit the netplan generated NetworkManager config in a desktop GUI the updated config will be written back to netplan with a generated UUID and not touching the original file. This ment changes made by a user are lost after a reboot because the cloud-init netplan config will always be preferred and overrides the NM generated one. By triggering the save operation for every cloud-init generated config after first boot and then removing the original file this issue can be resolved.

Reply:
I'm curious if things are different on raspberry-pi-os, but on my desktop at least, I see the network manager files get written with a '90-' first, e.g., `90-NM-141e654c-21b3-4e52-9505-e3c94be335c0.yaml` so anything in that file would override what is in `50-cloud-init.yaml`. If it doesn't work this way, I would think this would be a netplan bug, or the patch was applied incorrectly. Is this something you can check to see if you get different results?